pkg/util/restmapper/restmapper_test.go

```diff
@@ -0,0 +1,271 @@
+package restmapper
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	discoveryfake "k8s.io/client-go/discovery/fake"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	coretesting "k8s.io/client-go/testing"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+var fakeResources = []*metav1.APIResourceList{
+	{
+		GroupVersion: appsv1.SchemeGroupVersion.String(),
+		APIResources: []metav1.APIResource{
+			{Name: "deployments", Namespaced: true, Kind: "Deployment"},
+		},
+	},
+	{
+		GroupVersion: schema.GroupVersion{
+			Group:   "",
+			Version: "v1",
+		}.String(),
+		APIResources: []metav1.APIResource{
+			{Name: "pods", Namespaced: true, Kind: "Pod"},
+		},
+	},
+	{
+		GroupVersion: schema.GroupVersion{
+			Group:   "",
+			Version: "v2",
+		}.String(),
+		APIResources: []metav1.APIResource{
+			{Name: "pods", Namespaced: true, Kind: "Pod"},
+		},
+	},
+	{
+		GroupVersion: schema.GroupVersion{
+			Group:   "extensions",
+			Version: "v1beta",
+		}.String(),
+		APIResources: []metav1.APIResource{
+			{Name: "jobs", Namespaced: true, Kind: "Job"},
+		},
+	},
+}
+
+// sum = 10 repeat = 6
+var kindTCs = []struct {
+	want  schema.GroupVersionResource
+	input schema.GroupVersionKind
```

Comment:
We can organize the cases, like:
```golang
// getGVRTestCases organize the test cases for GetGroupVersionResource.
// It can be shared by both benchmark and unit test.
var getGVRTestCases = []struct {
	name        string
	inputGVK    schema.GroupVersionKind
	expectedGVR schema.GroupVersionResource
	expectErr   bool
}{
	{},
}
```